framework/PostProcessors/Metric.py

```diff
@@ -0,0 +1,293 @@
+
+    unidentifiedVars = set(self.features) - set(inputDict['features'].keys())
+    if len(unidentifiedVars) > 0:
+      self.raiseAnError(IOError, "Features: '", str(unidentifiedVars), "' could not be found in the input objects")
+    unidentifiedVars = set(self.targets) - set(inputDict['targets'].keys())
+    if len(unidentifiedVars) > 0:
+      self.raiseAnError(IOError, "Targets: '", str(unidentifiedVars), "' could not be found in the input objects")
+
+    return inputDict
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the pp.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+    self.__workingDir = runInfo['WorkingDir']
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs
+      @ In, xmlNode, xml.etree.ElementTree Element Objects, the xml element node that will be checked against the available options specific to this Sampler
+      @ Out, None
+    """
+    paramInput = Metric.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+
+    for child in paramInput.subparts:
+      if child.getName() == 'Metric':
+        if 'type' not in child.parameterValues.keys() or 'class' not in child.parameterValues.keys():
+          self.raiseAnError(IOError, 'Tag Metric must have attributes "class" and "type"')
+        else:
+          metricName = child.value.strip()
+          self.metricsDict[metricName] = None
+      elif child.getName() == 'Features':
+        self.features = list(var.strip() for var in child.value.split(','))
+        self.featuresType = child.parameterValues['type']
+      elif child.getName() == 'Targets':
+        self.targets = list(var.strip() for var in child.value.split(','))
+        self.TargetsType = child.parameterValues['type']
+      else:
+        self.raiseAnError(IOError, "Unknown xml node ", child.getName(), " is provided for metric system")
+
+    if not self.features:
+      self.raiseAnError(IOError, "XML node 'Features' is required but not provided")
+    elif len(self.features) != len(self.targets):
+      self.raiseAnError(IOError, 'The number of variables found in XML node "Features" is not equal the number of variables found in XML node "Targets"')
+
+  def collectOutput(self,finishedJob, output):
+    """
+      Function to place all of the computed data into the output object, (Files or DataObjects)
+      @ In, finishedJob, object, JobHandler object that is in charge of running this postprocessor
+      @ In, output, object, the object where we want to place our computed results
+      @ Out, None
+    """
+    if finishedJob.getEvaluation() == -1:
```

Comment:
This is the old way (if you copied paste from another PP, also that should be changed). 
it should be something like

```
    evaluation = finishedJob.getEvaluation()
    if isinstance(evaluation, Runners.Error):
      self.raiseAnError(RuntimeError,"No available Output to collect")
```

Reply:
Changed.

Reply:
It seems that most of PP are using the old style. Let me change that.

Reply:
Thanks a lot.